lib/guest_installation_and_configuration_base.pm

```diff
@@ -772,7 +772,12 @@ sub config_guest_network_bridge_device {
         if ($self->{guest_netaddr} ne 'host-default') {
             $self->write_guest_network_bridge_device_config($_bridge_network, $_bridge_device, 'static', 'auto', '', 'yes');
             my $_bridge_device_config_file = '/etc/sysconfig/network/ifcfg-' . $_bridge_device;
-            script_retry("wicked ifup $_bridge_device_config_file $_bridge_device", retry => 3, die => 0);
+            script_retry("wicked ifup $_bridge_device_config_file $_bridge_device", retry => 3, die => 0) if is_sle;
+            #Not sure if it is a bug, but it works in openSUSE this way.
+            if (is_opensuse) {
+                my $_bridge_ipaddr = script_output ("grep IPADDR $_bridge_device_config_file | cut -d \"'\" -f2");
+                script_retry("brctl addbr $_bridge_device; ip addr add $_bridge_ipaddr dev $_bridge_device; ip link set $_bridge_device up", retry => 3, die => 0);
+            }
```

Comment:
This part can be further improved as below:

if ($self->${guest_os_name} ne 'opensuse') {
    script_retry wicked
}
else {
   script_retry brctl
}

Reply:
@waynechen55 this is very hard to read. Can you please use formatting in your comments?

Reply:
Should be only change to use if/else. Done.

Reply:
Could you help add comments here to explain why opensuse use brctl instead of wicked ?

Reply:
I have not found the reason yet. The document of openSUSE Leap describes both brctl and wicked are supported, but I have not found some official document of Tumbleweed yet. I'll begin to open bugs about all the issues in TW which I found in the test, such as libvirtd(or $PATH), br0 and this issue(which may have lower priority).

Reply:
It seems that is_opensuse and is_sle are not recommended by oliver ? I can help make this part better if is_opensuse and is_sle are really not recommended.

Reply:
No, I did not say brctl makes nic up while wicked does not. I checked wicked.server is running by default in TW. but neither 'wicked ifup' nor 'ifup' can make the interface get an IP from dhcp server. the device is up but no ip. It seems the network configuration file did not take effect(network restart did not take effect either). 

just brctl did not work. 'brctl addbr' + 'ip addr add' + 'ip link set up' must be used in a row. I guess 'brctl' can be replaced with 'ip' command too(I did not try).

Reply:
So the if (is_opensuse) branch really works for your test but the other branch(using wicked) does not, right ?

Reply:
correct.

Reply:
> 2) If this is temporary solution, please also add some comments.
It is ok.

> 3)It seems that is_opensuse and is_sle are not recommended by oliver ? I can help make this part better if is_opensuse and is_sle are really not recommended.

Thanks. Let's see if it is a bug or if there will be a fix.